locallayout/v1_facade.go

```diff
@@ -212,43 +224,41 @@ func newEmptyLayerListFrom(configFile *v1.ConfigFile) []v1.Layer {
 }
 
 func (l v1LayerFacade) Digest() (v1.Hash, error) {
-	return l.diffID, nil
+	if l.optionalUnderlyingLayer != nil {
+		return l.optionalUnderlyingLayer.Digest()
+	}
+	return v1.NewHash("sha256:90e01955edcd85dac7985b72a8374545eac617ccdddcc992b732e43cd42534af")
```

Comment:
This and `Size()` below is the big bummer - we can't construct a valid manifest without it. But, we can't know it without pulling all the layers from the daemon. Many sighs